source/agora/node/FullNode.d

```diff
@@ -1205,27 +1205,32 @@ public class FullNode : API
 
     private void pushBlock (const Block block) @trusted
     {
-        if (this.is_shutting_down)
-            return;
         const now = StdClock.currTime();
         foreach (index, ref handler; this.block_handlers)
         {
+            if (this.is_shutting_down)
+                return;
             if (handler.disabledUntil > now)
                 continue;
 
             this.taskman.runTask({
                 // Work around potential DMD bug
                 const idx = index;
+                if (this.block_handlers.length <= idx)
+                    return;
+
+                auto _handler = this.block_handlers[idx];
+
```

Comment:
You can remove these other changes now as if not shutting down the array will not be null.

Reply:
It's for accessing the `this.block_handlers` after the variable has been set to null which means the handler is an empty array. This is kind of related to #2942.